src/aurelia-slickgrid/custom-elements/aurelia-slickgrid.ts

```diff
@@ -160,50 +164,51 @@ export class AureliaSlickgridCustomElement {
 
   constructor(
     private readonly aureliaUtilService: AureliaUtilService,
-    private readonly bindingEngine: BindingEngine,
-    private readonly container: Container,
-    private readonly elm: HTMLDivElement,
-    private readonly globalEa: EventAggregator,
+    @IObserverLocator private readonly observerLocator: IObserverLocator,
+    @IContainer private readonly container: IContainer,
+    private readonly elm: Element,
+    @IEventAggregator private readonly globalEa: IEventAggregator,
     private readonly containerService: ContainerService,
     private readonly translaterService: TranslaterService,
-    externalServices: {
-      backendUtilityService?: BackendUtilityService,
-      collectionService?: CollectionService,
-      eventPubSubService: EventPubSubService,
-      extensionService?: ExtensionService,
-      extensionUtility?: ExtensionUtility,
-      filterService?: FilterService,
-      gridEventService?: GridEventService,
-      gridService?: GridService,
-      gridStateService?: GridStateService,
-      groupingAndColspanService?: GroupingAndColspanService,
-      paginationService?: PaginationService,
-      resizerService?: ResizerService,
-      rxjs?: RxJsFacade,
-      sharedService?: SharedService,
-      sortService?: SortService,
-      treeDataService?: TreeDataService,
-    }
+    // TODO: MB - not sure what this is for
+    // externalServices: {
```

Comment:
this `externalServices`, as mentioned in our discussion, was used only for my Jest Unit Tests since I was instantiating the custom element directly (basically without DI, it's easier for me to test when instantiating manually) and in Angular-Slickgrid I use `@Inject('externalServices')`, does Au2 have anything similar to inject something custom without necessarily passing the usual DI? Instantiating my own arguments also allows me to pass mock services more easily than using the typical Aurelia testing structure which I had problem with in Au1 at least.

Reply:
We can define an interface and inject it similar to `IContainer` for example

Reply:
Actually, I would just inject those services and let DI create them. You can provide mocks during testing as before.